### DISCUSSION TITLE
Image optimization/proxy after upload

### DISCUSSION THREAD

**Discussion by furkankarayel** [pocketbase/pocketbase#4942] [Q&A] [answered]
Hi,

I'm running Pocketbase as the backend of a webshop system and I would like to optimize the images which cannot be optimized, as they're not stored locally. 
My images are currently stored on Cloudflare R2 Storage and I serve my frontend over Cloudflare Pages.

I would like to build a feature for my case and possibly share it with the community. I'd like to run an image proxy on my VPS and serve them for my frontend. 
But for being able to do that I'd need to know which parts are responsible for the image upload. 

Where should I take a look at to implement an usage of image proxy/optimiziation after uploading images via Pocketbase?

Kind regards,
Furkan

**Comment by thangngoc89** [comment]
Are you going to write the image optimization part into the Pocketbase's binary or JS's hook or are you going to call some kind of backend and only need the link of the images on Cloudflare R2?

**Reply by furkankarayel** [reply]
> Are you going to write the image optimization part into the Pocketbase's binary or JS's hook or are you going to call some kind of backend and only need the link of the images on Cloudflare R2? 

I'd like to implement in Pocketbase's binary so by doing that I could store the images locally on my VPS. The next step I'd take is possibly generating different files and different dimensions so they could be used on different screen sizes. 
On client side I might write the logic to choose the right image dimensions for the screen sizes. 
Could you recommend me anything, as it's just an initial idea how I would like to solve it.

**Reply by thangngoc89** [reply, marked as answer]
@furkankarayel What you’re describing is already available in Pocketbase. See documentation: https://pocketbase.io/docs/files-handling/#file-url

Regarding S3 vs your VPS. Inside Pocketbase, S3 is treated as another filesystem so that you don’t need to worry where files are actually stored

**Comment by furkankarayel** [comment]
Oh wow, that would be a huge advantage for me already. Thank you very much!
